pkg/storage/endpoint/maintenance.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2025 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package endpoint provides storage operations for maintenance tasks.
+// This implementation follows the design specified in RFC-0118:
+// https://github.com/tikv/rfcs/blob/master/text/0118-pd-maintenance-endpoints.md
+//
+// The storage layer provides atomic operations to ensure only one maintenance
+// task can be active at a time, preventing Raft quorum loss during TiKV maintenance.
+
+package endpoint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tikv/pd/pkg/storage/kv"
+	"github.com/tikv/pd/pkg/utils/keypath"
+)
+
+// MaintenanceTask represents a maintenance task stored in etcd.
+// NOTE: This type is exported by HTTP API. Please pay more attention when modifying it.
+type MaintenanceTask struct {
+	Type           string `json:"type"`            // The type of maintenance task
+	ID             string `json:"id"`              // Unique identifier for the task
+	StartTimestamp int64  `json:"start_timestamp"` // Unix timestamp when the task started
+	Description    string `json:"description"`     // Optional description of the task
+}
+
+// MaintenanceStorage defines the interface for maintenance task storage operations.
+type MaintenanceStorage interface {
+	// LoadMaintenanceTask loads a maintenance task by type from storage.
+	LoadMaintenanceTask(taskType string) (*MaintenanceTask, error)
+	// LoadAllMaintenanceTasks loads all maintenance tasks from storage.
+	LoadAllMaintenanceTasks(f func(k, v string)) error
```

Comment:
Why need to input one function? It's better to return all tasks and hide the function in the implementation.